### PR TITLE
Handling Imbalanced Datasets in Machine Learning

### PR DESCRIPTION
**Pull request by collins-a**
Article as index.md.
Hero image as hero.jpg.
Other image; bagging.png.

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by collins-a** [comment]
This fixes #645

File: articles/imbalanced-data-in-ml/index.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+For classification problems, there exists a method known as [Tomek links](http://www.samdrazin.com/classes/een548/project1report.pdf). It aims to improve the accuracy of the classification of data. This is by removing as much class label noise as possible. It also works to remove borderline examples with a higher probability of being incorrect. This is known as Tomek link removal. Tomek links are points that are the closest neighbors to each other but have different class labels. Since this technique makes it possible to identify points near different class labels, it is easy to get rid of them until none are left. In the context of undersampling, this technique gets rid of unwanted overlap between classes. The result is having only neighbors of the same class in close proximity.
```
Comment (lalith1403) [inline articles/imbalanced-data-in-ml/index.md:50]: Can you define class label noise? How is it different from support vector machines? Even SVMs try to maximize the distance between the decision boundary and the support vectors.

Comment (lalith1403) [reply articles/imbalanced-data-in-ml/index.md:50]: If similar, can support vectors act as tomek links? If different how?

File: articles/imbalanced-data-in-ml/index.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+Thus, when dealing with imbalanced datasets, it is wise to use the correct evaluation metrics. It is not advisable to use accuracy as a measure of performance. We may consider metrics such as F1-score, precision, and recall.
+
+Here is an [article](https://towardsdatascience.com/metrics-to-evaluate-your-machine-learning-algorithm-f10ba6e38234) that defines the below metrics in greater detail.
+
+As defined in my previous [article](https://www.section.io/engineering-education/evaluating-ml-model-performance/):
+
+Precision is the number of true positives against the total positive results predicted by a classifier.
+
+Precision = True Positives / (True Positives + False Positives)
```
Comment (lalith1403) [inline articles/imbalanced-data-in-ml/index.md:80]: Consider wrapping it in $$ Precision = \frac{True Positives}{True Positives + False Positives} $$ It makes it more professional. Consider doing the same for all the formulas mentioned.

Comment (lalith1403) [reply articles/imbalanced-data-in-ml/index.md:80]: Math JAX file is needed at the end. @hectorkambow Your help is needed with this once the edits are made.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
A few minor changes and this should be good to go. Cheers!

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by collins-a** [comment]
@lalith1403 changes made. Thank you for direction regarding wrapping the formulas. Cheers, sir!

**Review by lalith1403 — APPROVED** [review]

**Comment by lalith1403** [comment]
@hectorkambow, the mathjax script might be needed. Can you also paste the same here to paste the script myself from next time whenever required?

**Comment by hectorkambow** [comment]
> @hectorkambow, the mathjax script might be needed. Can you also paste the same here to paste the script myself from next time whenever required?

I've actually pinned it in our convo in Slack- thats where I copy it from 🚀

**Pull request merged**